### PR TITLE
Add Integration Tests Failure Section

### PR DESCRIPTION
**Pull request by appuk**
Issue: https://github.com/ansible/tower-dashboard/issues/27

This PR contains the initial setup and a basic table displaying test failures across all jobs.
The table also contains filtering options as follows:
Screen Shot 2020-02-11 at 10 43 09 AM

Checklist:
- [x] Create table and endpoint to GET and POST to table "integration_tests" in database: `/jenkins/integration_tests`
- [x] Basic UI endpoint: `/jenkins/integration_test_results`
- [x] Tweek integration.groovy file to send the test results (results.xml) to the towerdashboard endpoint
- [ ] TODO: get feedback from some UI developer to make the code more efficient/user friendly

### CODE REVIEW

**Comment by kdelee** [comment]
https://media.giphy.com/media/11qAyKz9AbFEYM/giphy.gif

File: towerdashboard/jenkins/base.py [hunk 1]
```diff
@@ -200,6 +200,112 @@ def serialize_issues(project):
...
+                            payload['platform'], payload['bundle'], payload['tls'],
+                            payload['fips'], payload['ansible'], payload['status'], payload['url'])
+            db_access.execute(insert_query)
+            db_access.commit()
+
+        return flask.Response(
+                json.dumps({'Inserted': 'ok'}),
+                status=201,
+                content_type='application/json'
+            )
```
Comment (nixocio) [inline towerdashboard/jenkins/base.py]: Break line.

File: towerdashboard/jenkins/base.py [hunk 2]
```diff
@@ -200,6 +200,112 @@ def serialize_issues(project):
...
+        missing_keys = []
+        for key in required_keys:
+            if key not in payload:
+                missing_keys.append(key)
+        if missing_keys:
+            return flask.Response(
+            json.dumps({'Error': 'Missing required keys/value pairs for {}'.format(missing_keys)}),
+            status=400,
+            content_type='application/json'
+            )
```
Comment (kdelee) [inline towerdashboard/jenkins/base.py]: maybe we could refactor this to share code w/ the sign_off_jobs for testing that we have all required keys

Comment (kdelee) [reply towerdashboard/jenkins/base.py]: could accept the payload and required keys but have a default list

like 
```
def check_paylod(payload, required_keys=['name', 'tower', 'component', 'deploy', 'platform', 'bundle', 'tls', 'fips', 'ansible', 'status', 'url'])
   ....
```

File: towerdashboard/jenkins/base.py [hunk 3]
```diff
@@ -200,6 +200,112 @@ def serialize_issues(project):
...
+            if item.get(key):
+                delta = datetime.now() - datetime.strptime(
+                    item[key], '%Y-%m-%d %H:%M:%S'
+                )
+                item['freshness'] = delta.days
+        if discard_old:
+            items = [ x for x in items if x['freshness'] < 2 ]
+
+        return items
+
```
Comment (kdelee) [inline towerdashboard/jenkins/base.py]: wonder if we can move the `set_freshness` function from the other method that also defines it to a whole-file scope so we can share it

**Comment by kdelee** [comment]
@appuk feel free to merge when ready. I think in future I'd like to refactor the sign_off_jobs code to do similar thing to what you are doing as far as just creating jobs when we DO get results in.

I invite you to reach out to @dsesami to deploy....keep in mind this bug https://github.com/ansible/tower-dashboard/issues/39 which I would approve a PR to fix

**Pull request merged**